### PR TITLE
Fix speculative decoding bug

### PR DESCRIPTION
**Pull request by Romanticoseu**
## Description

IPEX_CPU optimized models have issues for speculative decoding with short prompts, and in warmup, `draf_model` will be deleted, so we should remove these attributes in `else

### 4. How to test?
- [x] Application test

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/speculative.py [hunk 1]
```diff
@@ -100,6 +100,12 @@ def generate(
                                          draft_model=self.draft_model,
                                          **new_speculative_kwargs)
     else:
+        # When `draft_model` is false, these attributes
+        # ralted to speculative decoding should be removed
```
Comment (qiyuangong) [inline python/llm/src/ipex_llm/transformers/speculative.py:104]: related

File: python/llm/src/ipex_llm/transformers/speculative.py [hunk 2]
```diff
@@ -100,6 +100,12 @@ def generate(
...
+        for var in ['max_step_draft', 'th_stop_draft', 'hf_adjust',
+                    'auto_th_stop_draft', 'auto_parameters', 'min_step_draft',
+                    'th_batch_num']:
+            value = kwargs.pop(var, None)
```
Comment (qiyuangong) [inline python/llm/src/ipex_llm/transformers/speculative.py:108]: If value is not used, remove value

File: python/llm/src/ipex_llm/transformers/speculative.py [hunk 3]
```diff
@@ -74,8 +74,7 @@ def generate(
             for var in ['max_step_draft', 'th_stop_draft', 'hf_adjust',
                         'auto_th_stop_draft', 'auto_parameters', 'min_step_draft',
                         'th_batch_num']:
-                value = kwargs.pop(var, None)
-            del self.draft_model
```
Comment (glorysdj) [inline python/llm/src/ipex_llm/transformers/speculative.py:78]: self.draft_model should keep for next round inference.

**Review by glorysdj — APPROVED** [review]
LGTM

**Pull request merged**